rainier-core/src/main/scala/com/stripe/rainier/compute/Gradient.scala

```diff
@@ -102,12 +102,8 @@ private object Gradient {
     def toReal: Real = child.op match {
       case LogOp => gradient.toReal * (Real.one / child.original)
       case ExpOp => gradient.toReal * child
-      case AbsOp =>
-        Real.lt(child.original,
-                Real.zero,
-                gradient.toReal * child.original / child,
-                Real.zero)
-      case NoOp => gradient.toReal
+      case AbsOp => gradient.toReal * child.original / child
```

Comment:
Hm, don't we still need to guard against child being zero?

Reply:
Yes you're right. I have changed it the last commit. It now does an equality comparison against zero. The original offender was the "lesser than" comparison.